include/swift/Driver/Driver.h

```diff
@@ -263,6 +265,53 @@ class Driver {
                           const ToolChain &TC, bool AtTopLevel,
                           JobCacheMap &JobCache) const;
 
+private:
+  void computeMainOutput(Compilation &C, const JobAction *JA,
+                         const OutputInfo &OI, const OutputFileMap *OFM,
+                         const ToolChain &TC, bool AtTopLevel,
+                         SmallVectorImpl<const Action *> &InputActions,
+                         SmallVectorImpl<const Job *> &InputJobs,
+                         const TypeToPathMap *OutputMap, StringRef BaseInput,
+                         llvm::SmallString<128> &Buf,
+                         CommandOutput *Output) const;
+
+  void chooseSwiftModuleOutputPath(Compilation &C, const OutputInfo &OI,
+                                   const OutputFileMap *OFM,
+                                   const TypeToPathMap *OutputMap,
+                                   CommandOutput *Output) const;
+
+  void chooseSwiftModuleDocOutputPath(Compilation &C,
+                                      const TypeToPathMap *OutputMap,
+                                      CommandOutput *Output) const;
+  void chooseRemappingOutputPath(Compilation &C, const TypeToPathMap *OutputMap,
+                                 CommandOutput *Output) const;
+
+  void chooseSerializedDiagnosticsPath(Compilation &C, const JobAction *JA,
+                                       const OutputInfo &OI,
+                                       const TypeToPathMap *OutputMap,
+                                       CommandOutput *Output) const;
+
+  void chooseDependenciesOutputPaths(Compilation &C, const OutputInfo &OI,
+                                     const TypeToPathMap *OutputMap,
+                                     llvm::SmallString<128> &Buf,
+                                     CommandOutput *Output) const;
+
+  void chooseSaveOptimizationPath(Compilation &C, const OutputInfo &OI,
```

Comment:
The noun here is "optimization record", not "save optimization".

Reply:
Ah, ok!